src/components/QuestionDisplay/QuestionDisplay.jsx

```diff
@@ -0,0 +1,72 @@
+import {
+  Box,
+  Card,
+  CardContent,
+  LinearProgress,
+  Typography,
+} from '@mui/material';
+import AnswerButton from './AnswerButton';
+import useDecodeHtmlEntities from '../../hooks/useDecodeHtmlEntities';
+import useQuestionDisplay from '../../hooks/useQuestionDisplay';
+
+function QuestionDisplay() {
+  const {
+    currentQuestion,
+    shuffledAnswers,
+    selectedAnswer,
+    showResult,
+    handleAnswer,
+    progress,
+    quizState,
+  } = useQuestionDisplay();
+
+  const decodedQuestion = useDecodeHtmlEntities(currentQuestion.question);
+  const decodedShuffledAnswers = shuffledAnswers.map((answer) => useDecodeHtmlEntities(answer));
```

Comment:
`useDecodeHtmlEntities` is not a hook so this is not the case but…
Don't call hooks in the loop. This loop could've been part of the hook logic… if this would be a hook of course